src/types.jl

```diff
@@ -369,14 +370,14 @@ function ReplicaSystem(;
                 n_replicas,
                 boundary,
                 neighbor_finder=NoNeighborFinder(),
```

Comment:
Right now I am assuming neighbor finder to me immutable, so that same one can be passed to all the replicas. But the `CellListNeigbourFinder` won't work as it stores some data, as @jgreener64 mentioned earlier. Maybe the the neighbor finders can also be passed like loggers are being passed as a Tuple?

Reply:
Yes sounds good.